Framework/Types/inc/MantidTypes/Core/DateAndTime.h

```diff
@@ -147,10 +147,66 @@ class MANTID_TYPES_DLL DateAndTime {
 private:
   /// A signed 64-bit int of the # of nanoseconds since Jan 1, 1990.
   int64_t _nanoseconds;
+
+  /// Max allowed nanoseconds in the time; 2^62-1
+  static constexpr int64_t MAX_NANOSECONDS = 4611686018427387903LL;
```

Comment:
This can't be done with `std::numeric_limits::max` presumably? not sure why this isn't 2^63-1?

Reply:
This is unrelated to this PR (just moved code). The reason seems to be to allow addition that does not overflow.